### PR TITLE
feat: Added support for v1_20_R4+ and nullable spawners

### PR DESCRIPTION
**Pull request by Gertius**
Resolves https://github.com/CorneliusMa/SilkSpawners_v2/issues/50
Resolves https://github.com/CorneliusMa/SilkSpawners_v2/issues/52
Resolves https://github.com/CorneliusMa/SilkSpawners_v2/issues/53

### CODE REVIEW

**Comment by Gertius** [comment]
https://github.com/CorneliusMa/SilkSpawners_v2/pull/51/commits/542fda4437c07ad8a07e8c1ec55a934bccf26de3 now also resolves https://github.com/CorneliusMa/SilkSpawners_v2/issues/52

**Comment by jurohn** [comment]
@Gertius Can you build this for us as a beta jar?

**Comment by Gertius** [comment]
> @Gertius Can you build this for us as a beta jar?

[SilkSpawners_v2.zip](https://github.com/CorneliusMa/SilkSpawners_v2/files/15256038/SilkSpawners_v2.zip)

**Comment by Christoffyw** [comment]
Yes please

File: v1_20_R4/src/main/java/de/corneliusmay/silkspawners/nms/v1_20_R4/NMSHandler.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.bukkit.attribute.AttributeInstance;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.util.RayTraceResult;
+
+public class NMSHandler implements NMS {
+
+    @Override
+    public Block getTargetBlock(Player player) {
```
Comment (CorneliusMa) [inline v1_20_R4/src/main/java/de/corneliusmay/silkspawners/nms/v1_20_R4/NMSHandler.java:15]: Could you please explain the reasoning behind replacing return player.getTargetBlockExact(5); with the new code block? Specifically, I am not clear on the use of RayTraceResult and its benefits over the original method. A detailed explanation would be appreciated.

File: v1_20_R4/src/main/java/de/corneliusmay/silkspawners/nms/v1_20_R4/NMSHandler.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        double range = 5;
+
+        AttributeInstance blockRange = player.getAttribute(Attribute.PLAYER_BLOCK_INTERACTION_RANGE);
+        if(blockRange != null) {
+            range = blockRange.getValue();
+        }
+
+        RayTraceResult hitResult = player.rayTraceBlocks(range);
+        return hitResult != null ? hitResult.getHitBlock() : null;
+    }
```
Comment (CorneliusMa) [inline v1_20_R4/src/main/java/de/corneliusmay/silkspawners/nms/v1_20_R4/NMSHandler.java:25]: Could you please explain the reasoning behind replacing return player.getTargetBlockExact(5); with the new code block? Specifically, I am not clear on the use of RayTraceResult and its benefits over the original method. A detailed explanation would be appreciated.

**Review by rxchard — APPROVED** [review]
LGTM

**Pull request merged**